src/callbacks.c

```diff
@@ -475,6 +475,37 @@ void on_entry_tagfilter_activate(GtkEntry *entry, gpointer user_data)
 }
 
 
+void on_entry_docfilter_changed(GtkAction *action, gpointer user_data)
+{
+	GeanyDocument *doc = document_get_current();
+	GtkEntry *filter_entry;
+
+	if (!doc)
+		return;
+
+	filter_entry = GTK_ENTRY(ui_lookup_widget(main_widgets.window, "entry_docfilter"));
+	sidebar_openfiles_set_filter(gtk_entry_get_text(filter_entry));
+}
+
+
+void on_entry_docfilter_icon_press(GtkEntry *entry, GtkEntryIconPosition icon_pos, GdkEvent *event, gpointer user_data)
```

Comment:
We could save this callback if we use `ui_entry_add_clear_icon()` which does the same and is just a convenience way of doing this.

But since it doesn't add any functionality, we can also leave it as is.

Reply:
I'll probably leave it as it is for now.

I'm planning to modify the symbol tree filtering feature to also use `GtkTreeModelFilter` (it doesn't work this way now, filtering is done "manually" which doesn't preserve the tree structure) and I could switch to using `ui_entry_add_clear_icon()` for both of them.